### PR TITLE
snap-recovery: add minimal binary so that we can use spread on it

### PR DESCRIPTION
**Pull request by mvo5**
This adds a minimal snap-recovery binary and spread spread tests for creating an initial partition and append to an existing partition.

Based on https://github.com/snapcore/snapd/pull/7609

### CODE REVIEW

File: gadget/partition.go [hunk 1]
```diff
@@ -113,3 +113,31 @@ func runSfdisk(image string, script string) error {
 	}
 	return nil
 }
+
+// PositionedVolumeFromGadget takes a gadget rootdir and positions the
+// partitions as specified.
+func PositionedVolumeFromGadget(gadgetRoot string) (*LaidOutVolume, error) {
```
Comment (bboozzoo) [inline gadget/partition.go]: This should probably be moved to layout.go or even gadget.go

File: cmd/snap-recovery/partition/sfdisk.go [hunk 2]
```diff
@@ -100,18 +100,14 @@ func (sf *SFDisk) Create(pv *gadget.LaidOutVolume) (map[string]string, error) {
...
+	// Write the partition table, note that sfdisk will re-read the
+	// partition table by itself: see disk-utils/sfdisk.c:write_changes()
 	cmd := exec.Command("sfdisk", sf.device)
 	cmd.Stdin = buf
 	if output, err := cmd.CombinedOutput(); err != nil {
 		return deviceMap, osutil.OutputErr(output, err)
 	}
 
-	// Reload the partition table using blockdev which is part of the initramfs
-	if output, err := exec.Command("blockdev", "--rereadpt", sf.device).CombinedOutput(); err != nil {
```
Comment (bboozzoo) [inline cmd/snap-recovery/partition/sfdisk.go]: Right, this only makes sense when we operate on the actual device and want to make the kernel reload the partition table. At some point perhaps osutil would even be a better place for this.

Comment (mvo5) [reply cmd/snap-recovery/partition/sfdisk.go]: Yeah, we don't need it here, I checked the sfdisk code and it is doing this by itself after it has written the changes to the disk.

Comment (mvo5) [reply cmd/snap-recovery/partition/sfdisk.go]: I also added a comment close to the sfdisk call that this already rescanns the partition table.

**Pull request merged**